protostar/commands/cairo1_commands/fetch_from_scarb.py

```diff
@@ -124,3 +125,12 @@ def maybe_fetch_linked_libraries(project_root_path: Path) -> Optional[List[Path]
         raise ScarbMetadataFetchException("Error parsing metadata:\n" + str(ex)) from ex
 
     return paths
+
+
+def find_directory_with_cairo_project_toml(lib_cairo_path: Path) -> Path:
+    src_directory_path = lib_cairo_path.parent
+    return (
+        src_directory_path
+        if "cairo_project.toml" in os.listdir(src_directory_path)
```

Comment:
why??

Reply:
Path should point to the location with cairo_project.toml - some libraries like [quaireaux](https://github.com/keep-starknet-strange/quaireaux/tree/main/quaireaux/math) have cairo_project.toml outside of the src folder.

Reply:
`cairo_project.toml` does not have to match the project structure, it's merely a set of arguments for the compiler saved to a file, and anyway it is a dependency, so you shouldn't touch its structure at all -- all the paths are provided by Scarb already

Reply:
As discussed offline - we have to do it this way until changes are made in the cairo fork.
Issue: https://github.com/software-mansion/protostar/issues/1704